### ISSUE TITLE
Missing key "subject_encoder" in the loaded_state_dicts from pretrained model "epoch_8.pth"

### ISSUE THREAD

**Issue by MW-ZHOU** [LilyDaytoy/OpenPVSG#7]
Hi,

I am running the file `rel_test_full.py` to do some testing, but I got a KeyError: "subject_encoder" when accessing the `loaded_state_dicts` which is loaded from the pre-trained model `epoch_8.pth`,  the variable `save_work_dir` is set to be
 `work_dirs/relation/rel_ips_transformer`, which contains the model `epoch_8.pth`

as far as I can tell, we are supposed to load the model from `epoch_8.pth` which is under the folder "mask2former_r50_ips"

Can you please help me with this? Please point out any mistakes that I have made. Thanks

<img width="1470" alt="image" src="https://github.com/LilyDaytoy/OpenPVSG/assets/41655056/d502e52f-e88b-4d12-b99d-e26d699f68af">

**Comment by Jingkang50** [comment]
No, the model for rel_test are the second stage models, you can find them in the ckpt table in readme, such as the link below base, convolution, and transformer

**Comment by MW-ZHOU** [comment]
So I need to load the model from one of these `latest.pth` file under the relation folder, if I understand you correctly?
<img width="1634" alt="image" src="https://github.com/LilyDaytoy/OpenPVSG/assets/41655056/d1a08376-edec-402b-8e10-9a7a8894d049">

**Comment by MW-ZHOU** [comment]
The links below vanilla, filter, conv, and transformer all go to the same webpage.

**Comment by Jingkang50** [comment]
Please follow the entire process here: https://github.com/LilyDaytoy/OpenPVSG?tab=readme-ov-file#training-and-testing

Basically if you need test, you should save feature tube from the first stage with epoch_8.ckpt and run test with any provided models in the directory you mentioned.

**Comment by MW-ZHOU** [comment]
Can I skip the "Train IPS" and start with "Tracking and save query features"? 

Because I am currently also running `prepare_query_tube_ips.py`, which is used to save the feature tubes as you said, right? And I think it uses the `epoch_8.pth`, which is generated in the first stage.

**Comment by Jingkang50** [comment]
Yes, make sense to me.

**Comment by MW-ZHOU** [comment]
Thanks a lot for your time and help.

**Comment by Jingkang50** [comment]
No problem! Thank you for your interest in the project!

**Comment by MW-ZHOU** [comment]
One more question, I want to use the trained model from stage two to generate scene graph and extract relation features for another dataset, but this dataset does not have an annotation file like `pvsg.json`. Is it possible to do this? Because as what I can see, `pvsg.json` is used to generate tube features and is used in every step for both training and testing.

**Comment by Jingkang50** [comment]
I think you might need to rewrite the `dataset` class. For inference, the pvsg.json only provides the video name, so I think you can modify some code for your task.

**Comment by MW-ZHOU** [comment]
Hi,
I wonder how to generate relation predictions for an entire video like you did in the image on the top of the page. Because when I run `rel_test_full.py`, it only generates one prediction for the video "0004_11566980553". What mistake did I make?
What am I missing?

![image](https://github.com/LilyDaytoy/OpenPVSG/assets/41655056/88ec26fb-7b1e-4415-b4e4-55b16868d836)

<img width="1567" alt="image" src="https://github.com/LilyDaytoy/OpenPVSG/assets/41655056/0bb3765f-b680-4b01-b5b0-9180831e5076">

**Comment by MW-ZHOU** [comment]
No worries, I have found the python file to do this.

**Comment by MW-ZHOU** [comment]
Hi,

I have a question about using the code from [PVSGAnnotation](https://github.com/LilyDaytoy/PVSGAnnotation). What does xxxxxx mean in the link? How do I use this link? Can you please help me with this?
<img width="1264" alt="image" src="https://github.com/LilyDaytoy/OpenPVSG/assets/41655056/d5c9f481-9e2f-4d72-87a5-1ac7425baca1">

**Comment by Jingkang50** [comment]
Oh yes, xxxx means a job id in superannotate. 
[Here](https://entuedu-my.sharepoint.com/:f:/g/personal/jingkang001_e_ntu_edu_sg/EqwyAOskPk9NobSP5IhAcGMB-df9kdeZjEE-AK4XPNzccA?e=ztFdyv) is an example you can play with.

**Comment by MW-ZHOU** [comment]
Sure, that would be great. How about I connect with you via email for later communication?

**Comment by qnngogogo** [comment]
@MW-ZHOU Hello, I would like to ask some questions. I'm also trying to execute prepare_query_tube_vps.py file at the moment. I have downloaded the vidor dataset but it can't run successfully, what file do I need to change? Or is it not right to execute the order. The command I'm using is YTHONPATH=./ CUDA_VISIBLE_DEVICES=0 python tools/prepare_query_tube_vps.py configs/mask2former_vps/mask2former_video_r50_single_video_test.py work_dirs/mask2former_ r50_vps/epoch_8.pth --work-dir work_dirs/mask2former_r50_vps --split val
